### PR TITLE
strv_split: pass EXTRACT_RETAIN_ESCAPE

### LINKED ISSUE #18952: fstab generated Requires= dependency on mount is missing escaping
**systemd version the issue has been seen with**
systemd 247.3 (though it's present in latest main)

**Expected behaviour you didn't see**
 Example contents of /etc/fstab:
```
LABEL=/mnt/d0   /mnt/d0         rtxfs x-systemd.requires=/dev/disk/by-partlabel/\\x2fmnt\\x2fd0 0 2
```
On v246 this correctly generates Requires= dependencies for /mnt/d0 that looks like:
```
Requires="dev-disk-by\\x2dpartlabel-\\x5cx2fmnt\\x5cx2fd0.device"
```

**Unexpected behaviour you saw**
On v247 the example incorrectly generates:
```
Requires="dev-disk-by\\x2dpartlabel-x2fmntx2fd0.device"
```

I've bisected this to 0645b83a40d1c782f173c4d8440ab2fc82a75006. cc @keszybz in case you have a quick fix. I can look at it tomorrow since it's bedtime here.

### PR DESCRIPTION
**Pull request by anitazha**
Fixes #18952

### CODE REVIEW

**Comment by anitazha** [comment]
Possibly an extension or alternative to https://github.com/systemd/systemd/pull/18955 but I'm not sure this is the right fix hence the discussion label.

File: src/basic/strv.h [hunk 1]
```diff
@@ -77,7 +77,7 @@ int strv_split_full(char ***t, const char *s, const char *separators, ExtractFla
 static inline char **strv_split(const char *s, const char *separators) {
         char **ret;
 
-        if (strv_split_full(&ret, s, separators, 0) < 0)
+        if (strv_split_full(&ret, s, separators, EXTRACT_RETAIN_ESCAPE) < 0)
```
Comment (keszybz) [inline src/basic/strv.h:80]: Hmm, indeed, this is interesting. If this is done, escaping of the separator cannot be done. I think *some* users of strv_split() expect it, while others not. It's something to investigate independently.

**Review by keszybz — COMMENTED** [review]
In #18952, I now added another commit to drop fstab_extract_values(). Apart from the bug that we're trying to fix, fstab_extract_values() doesn't support the syntax for escaping the separator. So fstab_extract_values()  needs to go.

File: src/basic/strv.h (hunk 1 above)
Comment (bluca) [reply src/basic/strv.h:80]: add a flags parameter maybe?

Comment (anitazha) [reply src/basic/strv.h:80]: Depends on how the prior `strv_split` with `strv_split_extract` behaved before https://github.com/systemd/systemd/commit/0645b83a40d1c782f173c4d8440ab2fc82a75006 I suppose.

**Comment by poettering** [comment]
Given that #18952 is fixed now, do we want to keep this one open?

**Comment by anitazha** [comment]
Let's close

**Pull request closed without merging**